### PR TITLE
attempt to further clarify addr_of docs

### PR DESCRIPTION
**Pull request by RalfJung**

### CODE REVIEW

File: library/core/src/ptr/mod.rs [hunk 1]
```diff
@@ -2071,21 +2071,26 @@ impl<F: FnPtr> fmt::Debug for F {
...
 ///
-/// The `expr` in `addr_of!(expr)` is evaluated as a place expression, but never loads
-/// from the place or requires the place to be dereferenceable. This means that
-/// `addr_of!(*ptr)` is defined behavior even if `ptr` is null, dangling, or misaligned.
-/// Note however that `addr_of!((*ptr).field)` still requires the projection to
-/// `field` to be in-bounds, using the same rules as [`offset`].
+/// The `expr` in `addr_of!(expr)` is evaluated as a place expression, but never loads from the
+/// place or requires the place to be dereferenceable. This means that `addr_of!((*ptr).field)`
+/// still requires the projection to `field` to be in-bounds, using the same rules as [`offset`].
+/// However, `addr_of!(*ptr)` is defined behavior even if `ptr` is null, dangling, or misaligned.
```
Comment (the8472) [inline library/core/src/ptr/mod.rs:2077]: Maybe this should be under a `# Safety` headline.

Comment (the8472) [reply library/core/src/ptr/mod.rs:2077]: I think the opsem thread shows that users aren't really aware of the place expression being a separate feature.

Comment (RalfJung) [reply library/core/src/ptr/mod.rs:2083]: I added section headers, not sure if you had anything beyond that in mind.

**Comment by the8472** [comment]
LGTM

@bors r+ rollup

**Pull request merged**